src/SeleniumLibrary/keywords/waiting.py

```diff
@@ -135,15 +134,26 @@ def wait_until_page_contains_element(self, locator, timeout=None,
 
         ``error`` can be used to override the default error message.
         """
-        self._wait_until(
-            lambda: self.find_element(locator, required=False) is not None,
-            "Element '%s' did not appear in <TIMEOUT>." % locator,
-            timeout, error
-        )
+        if is_noney(limit):
+            self._wait_until(
+                lambda: self.find_element(locator, required=False) is not None,
+                "Element '%s' did not appear in <TIMEOUT>." % locator,
+                timeout, error
+            )
+        else:
+            limit = int(limit)
+            self._wait_until(
+                lambda: len(self.find_elements(locator)) is limit,
+                'Page should have contained "{}" element(s),'
+                'but it did contain "{}" element(s).'
+                'using locator "{}" did not appear in <TIMEOUT>.'.format(
```

Comment:
By the way, next release will support Python 3.6+, which will support  [f-strings](https://www.python.org/dev/peps/pep-0498/), which are more easier to use.

Reply:
Oh, but I should change the CI first, which is not yet done. Please ignore.